### PR TITLE
Implemented pages 1 and 2 of mockup

### PR DESCRIPTION
**Pull request by Gkrumbach07**
## Related Issues and Dependencies

https://github.com/thoth-station/opendatahub-cnbi/issues/17

## This introduces a breaking change

- No

## This Pull Request implements

- convert add image modal to to a page
- add breadcrumbs
- add dropdown for different image adding methods
- update state management to use useReducer
- added more columns to image table
- validation blocks submit button
- error header when api fails to respond after submit (500 code)
- small changes to wording

### Description

This PR implements the first two pages of the mockup. Essentially it updates the BYON implementation so that is able to support multiple methods of image adding.

### CODE REVIEW

File: frontend/src/pages/BYONImages/AddNewImageForm.tsx [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+      <Breadcrumb className="breadcrumb">
+        <BreadcrumbItem>Settings</BreadcrumbItem>
+        <BreadcrumbItem to="/notebookImages">Notebook image settings</BreadcrumbItem>
+        <BreadcrumbItem>Add new image</BreadcrumbItem>
+      </Breadcrumb>
+      <ApplicationsPage
+        loaded
+        empty={false}
+        title="Add new image"
+        description="Add a new image to Red Hat Data Science
```
Comment (codificat) [inline frontend/src/pages/BYONImages/AddNewImageForm.tsx]: I believe this should be:
```suggestion
        description="Add a new image to Open Data Hub
```

but beyond that: is there a better place to set the product name globally? (thinking in moving this downstream later)

**Review by codificat — CHANGES_REQUESTED** [review]
This looks great, thanks!

I only have one comment about RHODS vs ODH

File: frontend/src/pages/BYONImages/AddNewImageForm.tsx (hunk 1 above)
Comment (Gkrumbach07) [reply frontend/src/pages/BYONImages/AddNewImageForm.tsx]: that is the only place red hat data science is mentioned. Although we could set some global variable, it is not needed if it isnt used much

**Comment by codificat** [comment]
#3 adds CI configuration, that should let the required pre-commit run and allow me to `/approve`. Once that merges, a rebase should help us push this one forward. I'd take the opportunity to squash the 3 commits into one (or at least the last one with 7b28e816309721011cd0bc735a67017c809b1aab)

**Comment by Gkrumbach07** [comment]
replaced by https://github.com/thoth-station/odh-dashboard/pull/4

**Pull request closed without merging**